### DISCUSSION TITLE
Test classes end up in the packaged JAR

### DISCUSSION THREAD

**Discussion by LorenzoBettini** [eclipse-tycho/tycho#2001] [Q&A] [answered]
Following https://github.com/eclipse-tycho/tycho/blob/master/RELEASE_NOTES.md#240 I can successfully run plain JUnit tests in an eclipse-plugin project. As instructed there, I marked the folder containing tests as test source folder. However, its contents end up in the JAR because they are compiled both in target/classes (together with the main code) and in target/test-classes (where they should be). Do I have to specify some additional configuration to avoid that?

I'm using Tycho 2.7.5.

**Comment by laeubi** [comment]
> I'm using Tycho 2.7.5.

please try Tycho 4.0.0-SNAPSHOT where test handling was improved, if the problem is fixed there.

If **not** it would be good to provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue (e.g using the demo project as a basis:
https://github.com/eclipse-tycho/tycho/tree/master/demo/testing/surefire/with-source-folder

**Comment by LorenzoBettini** [comment]
I'll first try the demo project, but what I can immediately spot as a difference is that in https://github.com/eclipse-tycho/tycho/blob/master/demo/testing/surefire/with-source-folder/build.properties#L2 the `src_test` folder is not mentioned. Maybe that's what makes it work? In my project I have all the source folders (including the test ones) in build.properties

**Comment by laeubi** [comment, marked as answer]
If you put things in `build.properties` they are part of the "binary build" and thus included in the final jar.

**Comment by LorenzoBettini** [comment]
That was it! THANKS!
Sorry for the noise!
I seem to recall that PDE complains if a source folder is not in the build.properties (so I always add that); but it looks like it doesn't if it's a test source folder.

**Reply by laeubi** [reply]
Jes PDE is a bit strange in this regards, the whole `build.properties` story is actually a bit like maven-resource folder definition.

**Reply by LorenzoBettini** [reply]
Yes. In particular. It wants at least a source folder, but that's expected if you have binary output.
It doesn't complain about missing test source folders though, so I'm fine :)
Thanks again!
